### PR TITLE
fix getKey

### LINKED ISSUE #1228: Be safe with createFromContent
From sentry:
```
TypeError: Cannot read property 'getKey' of undefined
  at createWithContent (./universal/modules/outcomeCard/containers/OutcomeCard/OutcomeCardContainer.js:43:20)
...
```

this is actually thrown from within draft-js, so the best approach is a PR to facebook.
i'll have to play around with this to figure out where it fails that i can call .first() but not .getKey(). That's just weird!

### LINKED ISSUE #1243: Facilitator tethering issue
## Issue - Bug

When conducting 2nd session with a user group:
1) Facilitator advanced to 2nd teammate during 1st teammate's Social Check-In. (forgot we were all tethered to him)
2) Facilitator returned to 1st teammate but the other users had all become untethered
3) Facilitator attempted to return to beginning of meeting.
4) Other teammates hit refresh
5) Facilitator was kicked off of Facilitator role, which was reassigned to another teammate

6) Noted several times during subsequent meeting that teammate cards weren't always reflecting real-time changes during Updates section

7) At end of meeting, I watched new Facilitator End Meeting, however upon returning to my Dashboard, it appears the meeting has not been ended.

8) Team is unable to use Team Board b/c it's locked for "current meeting" :(

### PR DESCRIPTION
**Pull request by mattkrick**
This nukes bad records in the DB & makes it so no more bad records can enter the DB.
fixes #1243 fixes #1228

Problem:
The draft-js raw state had an empty block array. So, when you go to retrieve a block like `.first()` it will return undefined instead of an object. We could fix the rehydration function to always give it at least 1 block, but then the state on the client doesn't match what's in the DB, plus we'd have to be careful every time we did a rehydrate.

the simpler solution is to ensure that the blocks array has at least 1 block before writing it to the DB.

### CODE REVIEW

File: src/server/database/migrations/20170714185201-removeEmptyContent.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+exports.up = async (r) => {
+  r.table('Project')
+    .filter((doc) => doc('content').match('"blocks":\\[]'))
```
Comment (jordanh) [inline src/server/database/migrations/20170714185201-removeEmptyContent.js]: Chances are this regexp will be fine. I'm going to update it to be a little bit more robust. This pattern:

`"?blocks"?:\s*\\[\s*]`

Will make the quotes optional while also allowing zero or more characters of whitespace before the empty block (which also may contain zero or more characters of whitespace).

File: src/server/graphql/models/Project/projectMutation.js [hunk 2]
```diff
@@ -36,14 +31,15 @@ export default {
       // VALIDATION
       // TODO make id, status, teamMemberId required
       const schema = makeProjectSchema();
-      const {errors, data: validNewProject} = schema(newProject);
+      // ensure that content is not empty
+      const {errors, data: validNewProject} = schema({content: 1, ...newProject});
```
Comment (jordanh) [inline src/server/graphql/models/Project/projectMutation.js]: 👍

File: src/server/graphql/models/Project/projectMutation.js [hunk 3]
```diff
@@ -36,14 +31,15 @@ export default {
...
       handleSchemaErrors(errors);
 
       // RESOLUTION
       const now = new Date();
       const [userId] = validNewProject.teamMemberId.split('::');
       const {content} = validNewProject;
-      const {entityMap} = content ? JSON.parse(content) : convertToRaw(ContentState.createFromText(''));
+      const {entityMap} = JSON.parse(content);
```
Comment (jordanh) [inline src/server/graphql/models/Project/projectMutation.js]: Hey look, an optimization! I see what you did there :)

File: src/universal/validation/makeProjectSchema.js [hunk 4]
```diff
@@ -1,17 +1,35 @@
...
+        const keys = Object.keys(parsedContent);
+        if (keys.length !== 2 ||
+          typeof parsedContent.entityMap !== 'object' ||
+          !Array.isArray(parsedContent.blocks ||
+          parsedContent.blocks.length === 0)) {
+          return makeEmptyStr();
+        }
+        return str;
+      })
       .max(PROJECT_MAX_CHARS, 'Whoa! That looks like 2 projects'),
```
Comment (jordanh) [inline src/universal/validation/makeProjectSchema.js]: You know, somehow I never saw that validation error string before. Heh. I like it.

File: src/server/database/migrations/20170714185201-removeEmptyContent.js (hunk 1 above)
Comment (mattkrick) [inline src/server/database/migrations/20170714185201-removeEmptyContent.js]: oh no worries on that guy, that string is generated via JSON.stringify so it conforms to the exact spaceless, double quoted JSON standard

**Pull request merged**